File: Java/SpectableJavaExamplesTest/src/test/java/spectable/common/StatusTyped.java
package spectable.common;

import java.util.ArrayList;
import java.util.List;
import java.util.LinkedHashMap;
import java.util.Map;
import java.util.Objects;
import production.*;
import records.*;
import calculator.*;

public class StatusTyped {
    public int code;

    public StatusTyped(int code) {
        this.code = code;
    }

    public StatusTyped(StatusString s) {
        this.code = Integer.parseInt(s.code);
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof StatusTyped)) return false;
        StatusTyped that = (StatusTyped) o;
        return Objects.equals(code, that.code);
    }

    @Override
    public int hashCode() {
        return Objects.hash(code);
    }

    public StatusString toStatusString() {
        return new StatusString(String.valueOf(code));
    }

    public static List<StatusTyped> fromStringList(List<StatusString> list) {
        List<StatusTyped> result = new ArrayList<>();
        for (StatusString s : list) result.add(new StatusTyped(s));
        return result;
    }

    public static List<StatusString> toStringList(List<StatusTyped> list) {
        List<StatusString> result = new ArrayList<>();
        for (StatusTyped t : list) result.add(t.toStatusString());
        return result;
    }

    /** Internal plumbing for {@link Json}; use toJSON() for JSON text. */
    public Map<String, Object> toJsonValue() {
        Map<String, Object> m = new LinkedHashMap<String, Object>();
        m.put("code", code);
        return m;
    }

    public String toJSON() { return Json.write(toJsonValue()); }

    /** Internal plumbing for {@link Json}; use fromJSON(String) for JSON text. */
    public static StatusTyped fromJsonValue(Map<String, Object> m) {
        if (m == null) return null;
        return new StatusTyped(
            Json.asInt(Json.require(m, "code"), "code"));
    }

    public static StatusTyped fromJSON(String json) {
        return fromJsonValue(Json.parseObject(json));
    }

    public static String toJSONList(List<StatusTyped> list) {
        List<Object> arr = new ArrayList<Object>();
        if (list != null)
            for (StatusTyped item : list) arr.add(item == null ? null : item.toJsonValue());
        return Json.write(arr);
    }

    public static List<StatusTyped> fromJSONList(String json) {
        List<StatusTyped> result = new ArrayList<StatusTyped>();
        for (Object e : Json.parseArray(json))
            result.add(fromJsonValue(Json.asObject(e, "StatusTyped")));
        return result;
    }
}

File: CSharp/SpecStudioExamplesTestsCSharp/common/StatusTyped.cs
namespace SpecStudioExamplesTestsCSharp.common
{
using System;
using System.Buffers;
using System.Collections.Generic;
using System.Globalization;
using System.Text;
using System.Text.Json;
using production;

    public class StatusTyped
    {
        public int code;

        public StatusTyped(int code)
        {
            this.code = code;
        }

        public StatusString ToStatusString()
        {
            return new StatusString(
                Json.ToText(this.code)
            );
        }

        public static List<StatusString> ToStringList(List<StatusTyped> list)
        {
            var result = new List<StatusString>();
            foreach (var t in list) result.Add(t.ToStatusString());
            return result;
        }

        public static List<StatusTyped> FromStringList(List<StatusString> list)
        {
            var result = new List<StatusTyped>();
            foreach (var s in list) result.Add(s.ToStatusTyped());
            return result;
        }

        public void WriteJson(Utf8JsonWriter w)
        {
            w.WriteStartObject();
            w.WriteNumber("code", this.code);
            w.WriteEndObject();
        }

        public string ToJSON()
        {
            var buffer = new ArrayBufferWriter<byte>();
            using (var w = new Utf8JsonWriter(buffer)) { WriteJson(w); }
            return Encoding.UTF8.GetString(buffer.WrittenSpan);
        }

        public static StatusTyped FromJsonElement(JsonElement m)
        {
            return new StatusTyped(
                Json.AsInt(Json.Require(m, "code"), "code")
            );
        }

        public static StatusTyped FromJSON(string json)
        {
            return FromJsonElement(Json.Parse(json));
        }

        public static string ToJSONList(List<StatusTyped> list)
        {
            var buffer = new ArrayBufferWriter<byte>();
            using (var w = new Utf8JsonWriter(buffer))
            {
                w.WriteStartArray();
                foreach (var item in list) item.WriteJson(w);
                w.WriteEndArray();
            }
            return Encoding.UTF8.GetString(buffer.WrittenSpan);
        }

        public static List<StatusTyped> FromJSONList(string json)
        {
            var result = new List<StatusTyped>();
            var root = Json.Parse(json);
            Json.RequireArray(root, "StatusTyped");
            foreach (var e in root.EnumerateArray()) result.Add(FromJsonElement(e));
            return result;
        }

        public override string ToString()
        {
            return $"Code={code}";
        }

        public override bool Equals(object? obj)
        {
            if (obj is not StatusTyped other) return false;
            return object.Equals(this.code, other.code);
        }

        public override int GetHashCode()
        {
            var h = new System.HashCode();
            h.Add(this.code);
            return h.ToHashCode();
        }
    }
}
